### ISSUE TITLE
Identity is a post-detection problem: `agnostic_nms`, `per_class`, and an optional exclusivity pass over tracklets

### ISSUE THREAD

**Issue by Akseli-Ilmanen** [OCTRON-tracking/OCTRON-GUI#91]
> Follow-up to #87. Most of the analysis below was drafted with Claude. Please correct anything that misreads OCTRON.

## TL;DR

Training one YOLO class per individual works well for a single camera, but YOLO has no notion that individuals are mutually exclusive in space, and OCTRON's predict path currently cannot express that either. Three small things would close the gap, in increasing size:

1. Expose `agnostic_nms` on `octron predict`.
2. Document `per_class` (off by default, unsupported by some BoxMOT trackers) and recommend it for individual-as-class projects.
3. Optionally, a per-time-bin exclusivity pass over tracklets (`octron link`), and optionally a crop identity classifier trained from the organizer's instance IDs. Both fit in existing dependencies. I can also do these downstream over the CSVs but would be cool if handled directly within Octron.

## The problem

With individual-as-class, YOLOv8+ scores classes with independent sigmoids under a BCE loss, not a softmax. A box can score high for bird A *and* bird B at once, and nothing in training or inference says "bird B appears at most once per frame, and never at bird A's location". So when bird B is out of view, or never appears in a camera's ground truth (no negatives for that class there), the model can put a bird-B box on top of bird A. Exclusivity between individuals is a set-level decision and can only live after detection.

What the predict loop in `yolo_octron.py` does today:

- `agnostic_nms` is never passed to `model.predict`, so it is off. The bird-B-on-bird-A box survives per-class NMS and the tracking CSV reports two individuals at one location.
- `per_class` defaults to off in the tracker parameters, and several BoxMOT trackers (BoostTrack among them) do not support it, so a track can change class at a crossing.
- BoxMOT overwrites a track's class on each match; OCTRON maps each (BoxMOT track ID, label) pair to its own stable track ID. Good: a tracklet in the CSV is always label-consistent. Consequence: an identity flip mid-track becomes two tracklets rather than being resolved.
- `one_object_per_label` forces one track ID per label forever. If two animals are both called bird 3 they are merged into one track, not reassigned.

## Two ways to carry identity

**Individual as detector class** (what OCTRON does now, and a sensible default). One model, identity re-decided every frame, and cross-camera identity is automatic because every camera's detector uses the same class names. Costs: more labels per individual, the sigmoid issue above, and the detector must generalise appearance across views.

**Species detector + post-hoc identity.** One class ("bird") pooled across cameras; BoxMOT gives anonymous tracklets; a crop classifier (`yolo classify` on the organizer's crops, individual as class, trained from the organizer's **instance IDs**) is voted per tracklet. Crops are resized on their own, so a wide camera no longer sees the bird at half the pixels. The organizer needs no format change since label and instance are already separate. Optionally the classifier's embedding replaces OSNet as BoxMOT's ReID weights, which are pedestrian-trained and near useless for animals.

Neither works for unmarked identical conspecifics across non-overlapping cameras; then only count and transit-time constraints remain. I am curious if you have any suggestions that address both cases simultaneously!

## The piece that pays for itself in both: post-hoc exclusivity

Per time bin, build a matrix of tracklets alive in the bin (all synchronised videos) × known individuals, each cell the summed confidence of the frames in that tracklet that voted for that individual. This is readable off the tracking CSV as written today (top class + confidence per frame per track). Solve with hungarian assignment (`scipy.optimize.linear_sum_assignment`): one individual per tracklet, each individual at most once per bin. Greedy assignment lets two tracklets claim the same bird; the Hungarian solution gives the best joint assignment and hands the weaker tracklet its second-best identity. Cells where the chosen identity is not clearly above the alternatives are the junctions to surface for review.

`agnostic_nms=True` fixes a different failure (one animal, two boxes with two classes) and does not replace this step: two boxes on two different animals never overlap, so NMS cannot see the conflict.

## Proposal

1. `octron predict --agnostic-nms` (default off is fine).
2. A note on `per_class` in the tracker docs: what it guards against, which trackers support it.
3. Optional: `octron link <tracking.csv> [<tracking.csv> ...]` doing the exclusivity pass above; pandas + scipy only.
4. Optional: `octron train-identity` / `octron predict --identity weights.pt` for the crop classifier; ultralytics only.

Happy to test 3 and 4 on a four-camera rig and share labels!

**Comment by Akseli-Ilmanen** [comment]
For the Hungarian assignment step to be a bit more concrete, he is a worked example:

One time bin, two cameras, three known birds. Each cell is the summed confidence of that tracklet's frames voting for that bird:

| | bird 1 | bird 2 | bird 3 |
|---|---|---|---|
| tracklet A (cam 1) | 0.7 | 0.2 | 0.1 |
| tracklet B (cam 1) | 0.6 | 0.3 | 0.1 |
| tracklet C (cam 2) | 0.1 | 0.1 | 0.8 |

Taking each tracklet's top class alone gives A = bird 1 **and** B = bird 1, two birds at once in camera 1. The assignment picks A = bird 1, B = bird 2, C = bird 3 because that is the highest total under "each bird at most once". B's cell (0.3 vs 0.6) is the kind of weak decision to flag for review.

```python
from scipy.optimize import linear_sum_assignment
rows, cols = linear_sum_assignment(-scores)   # scores: (n_tracklets, n_individuals)
```
Inputs needed: the tracking CSVs (frame, track ID, class, confidence) and the list of individuals.
